service/src/main/java/uk/gov/hmcts/reform/fpl/service/docmosis/GeneratedOrderTemplateDataGeneration.java

```diff
@@ -40,22 +39,23 @@
     @Autowired
     private ChildrenService childrenService;
 
-    abstract DocmosisGeneratedOrderBuilder populateCustomOrderFields(CaseData caseData);
+    abstract DocmosisGeneratedOrder populateCustomOrderFields(CaseData caseData);
 
     @Override
     public DocmosisGeneratedOrder getTemplateData(CaseData caseData) {
         OrderTypeAndDocument orderTypeAndDocument = caseData.getOrderTypeAndDocument();
         GeneratedOrderType orderType = orderTypeAndDocument.getType();
 
-        DocmosisGeneratedOrderBuilder<?, ?> orderBuilder = populateCustomOrderFields(caseData);
+        DocmosisGeneratedOrder docmosisGeneratedOrder = populateCustomOrderFields(caseData);
+        var docmosisGeneratedOrderBuilder = docmosisGeneratedOrder.toBuilder();
 
```

Comment:
More minor things but I think that the names `order` and `orderBuilder` are slightly better here. Not sure if it is the best practice to name variables after their type, sometimes it makes sense to like with case data but this time I am not as sure.

Reply:
Could maybe merge these 2 lines as well. i.e.
``` java
var orderBuilder = populateCustomOrderFields(caseData).toBuilder();
```

Reply:
there is another orderBuilder in code base. I may leave it like this `docmosisGeneratedOrderBuilder` , what do you think? but i combined 2 lines as suggested.